packages/@react-spectrum/table/test/TreeGridTable.test.tsx

```diff
@@ -0,0 +1,1516 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+jest.mock('@react-aria/live-announcer');
+import {act, fireEvent, installPointerEvent, render as renderComponent, triggerPress, within} from '@react-spectrum/test-utils';
+import {announce} from '@react-aria/live-announcer';
+import {composeStories} from '@storybook/testing-react';
+import {enableTableNestedRows} from '@react-stately/flags';
+import {Provider} from '@react-spectrum/provider';
+import React from 'react';
+import {Scale} from '@react-types/provider';
+import * as stories from '../stories/TreeGridTable.stories';
+import {tableTests} from './Table.test';
+import {theme} from '@react-spectrum/theme-default';
+import userEvent from '@testing-library/user-event';
+
+let {
+  StaticExpandableRows: StaticExpandableTable,
+  DynamicExpandableRowsStory: DynamicExpandableTable,
+  ManyExpandableRowsStory: ManyRowsExpandableTable,
+  EmptyTreeGridStory: EmptyStateTable,
+  LoadingTreeGridStory: LoadingTable,
+  UserSetRowHeader: UserSetRowHeaderTable
+} = composeStories(stories);
+
+let onSelectionChange = jest.fn();
+let onExpandedChange = jest.fn();
+let onAction = jest.fn();
+
+let getCell = (tree, text) => {
+  // Find by text, then go up to the element with the cell role.
+  let el = tree.getByText(text);
+  while (el && !/gridcell|rowheader|columnheader/.test(el.getAttribute('role'))) {
+    el = el.parentElement;
+  }
+
+  return el;
+};
+
+let focusCell = (tree, text) => act(() => getCell(tree, text).focus());
+let moveFocus = (key, opts = {}) => {
+  fireEvent.keyDown(document.activeElement, {key, ...opts});
+  fireEvent.keyUp(document.activeElement, {key, ...opts});
+};
+
+let render = (children, scale = 'medium' as Scale, locale = 'en-US') => {
+  let tree = renderComponent(
+    <Provider theme={theme} scale={scale} locale={locale}>
+      {children}
+    </Provider>
+  );
+
+  act(() => {jest.runAllTimers();});
+  return tree;
+};
+
+let rerender = (tree, children, scale = 'medium' as Scale) => {
+  let newTree = tree.rerender(
+    <Provider theme={theme} scale={scale}>
+      {children}
+    </Provider>
+  );
+  act(() => {jest.runAllTimers();});
+  return newTree;
+};
+
+enableTableNestedRows();
+
+describe.only('TableView tests with expandable rows flag on', tableTests);
```

Comment:
So we don't actually want `UNSTABLE_hasExpandableRows` turned on for the table tests here since it would then render a TreeGridTableView instead of a TableView, resulting in test suite failures since the table tests look for `aria-rowindex` instead of `aria-posinset` etc. This `.only` essentially just want to turn the flag on but omit the `UNSTABLE_hasExpandableRows` prop and make sure that a typical TableView still renders.

I guess I could add checks in the base tableview tests to check that the treegrid role doesn't exist

Reply:
Oh maybe I misunderstood the intention. I assumed this was meant to test that the existing table stuff still passed when hasExpandableRows was true, but I guess you meant to test only that the feature flag didn't break anything? Once we delete the feature flag, this would just be removed entirely?

Reply:
correct, we can get rid of it when we delete the feature flag. As for testing the existing table behavior when `hasExpandableRows` is true, I've written tests for selection/keyboard nav/etc but ideally we'd pull some of the existing test cases from TableView and run them for a expandable row table, just need to figure out a good way to share those and refactor them to work for both base Table + expandable row